op-e2e/e2eutils/disputegame/helper.go

```diff
@@ -214,6 +219,18 @@ func (h *FactoryHelper) startOutputCannonGameOfType(ctx context.Context, l2Node
 	return NewOutputCannonGameHelper(h.T, h.Client, h.Opts, h.PrivKey, game, h.FactoryAddr, createdEvent.DisputeProxy, provider, h.System)
 }
 
+func (h *FactoryHelper) StartSuperCannonGameWithCorrectRoot(ctx context.Context, opts ...GameOpt) *SuperCannonGameHelper {
+	cfg := NewGameCfg(opts...)
+	require.NoError(h.T, wait.ForBlock(ctx, h.Client, 1))
+	b, err := h.Client.BlockByNumber(ctx, nil)
```

Comment:
This would return the unsafe head which isn't a valid proposal.  I think we'd want to use the current safe head timestamp.

I think it works because `WaitForSuperTimestamp` winds up waiting for it to become safe but probably better to just use the current safe head to start with.  The `wait.ForBlock` above would likely need to be `wait.ForSafeBlock` (or a variant that uses the EL instead of the rollupClient).

Reply:
fixed.